### PR TITLE
Add `fetch` to minic marathon management of URIs, i.e. fine managemen…

### PR DESCRIPTION
**Pull request by sheepkiller**
…t of mesos downloads.

### CODE REVIEW

**Comment by keis** [comment]
Code looks good to me. Could you update the README with some notes on how to use this?

**Comment by sheepkiller** [comment]
Hi @keis,

No problem, but my english sucks. 
BTW, when I write the patch, I wondered if I should keep the autoextraction of archive files.  My first thought was to enable it, but, since undefined boolean values are set to false, there's no way to figure out if the "don't extract flag" is deliberately set or not. Are you okay with this or we arbitrarily extract supported archive files ?

thanks

**Comment by alde** [comment]
I'd prefer to not automatically extract files

**Comment by sheepkiller** [comment]
hi @alde,
thank you for your clarifications.

**Comment by alde** [comment]
looks good to me

Could you squash the commits into one commit, and maybe reword the commit message slightly?
I'd put the explanation into the body of the message, and just keep the `Add 'fetch' to mimic marathon management of URIs` as the title.

**Comment by keis** [comment]
Thank you @sheepkiller :+1:
